src/core/sass/cache.ts

```diff
@@ -155,6 +156,21 @@ class SassCache {
       compressed,
     );
   }
+
+  // add a cleanup method to register a cleanup handler
+  cleanup(temp: TempContext | undefined) {
+    const registerCleanup = temp ? temp.onCleanup : onCleanup;
+    registerCleanup(() => {
+      try {
+        this.kv.close();
+        if (!temp) safeRemoveIfExists(this.path);
```

Comment:
What is this call to `safeRemoveIfExists` doing? I'm not sure I understand. We shouldn't remove that path.

Reply:
yes this is indeed wrong. Thanks for catching it ! 
I really meant 
```suggestion
        if (temp) safeRemoveIfExists(this.path);
```

We don't need to keep the sass cache dir when we create it as a temporary context. Though all TempContext and the Session temp `quarto-session*` folder will be deleted at some point by cleanup process. This is just a habit of making sure to remove from filesystem anything that code is creating, when not needed anymore. 

I understand you are suggesting to not do that for simplicity sake ?

Reply:
I just wanted to make sure that we're not deleting the system-wide cache that will be used when `useSessionCache` is false:

https://github.com/quarto-dev/quarto-cli/blob/6796ddfd42c87efbc1648b48125c742f12dd90c1/src/core/sass.ts#L307-L311

Reply:
All good then. Thanks for catching up my faulty condition !